controllers/ssp_controller.go

```diff
@@ -237,14 +240,20 @@ func (r *sspReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ct
 	return ctrl.Result{}, nil
 }
 
-func (r *sspReconciler) restartIfNeeded(sspObj *ssp.SSP) {
+func (r *sspReconciler) isRestartNeeded(sspObj *ssp.SSP) bool {
 	if reflect.DeepEqual(r.lastSspSpec, ssp.SSPSpec{}) {
-		return
+		return false
 	}
 	if !reflect.DeepEqual(r.lastSspSpec.TLSSecurityProfile, sspObj.Spec.TLSSecurityProfile) {
-		r.log.Info("TLSSecurityProfile changed, restarting")
-		os.Exit(0)
+		return true
 	}
+	return false
+}
+
+func (r *sspReconciler) restart(request *common.Request) {
+	r.log.Info("TLSSecurityProfile changed, restarting")
+	setSspResourceDeploying(request)
```

Comment:
Can you handle the error returned from `setSspResourceDeploying()`? It is enough to log it.

Reply:
sure good point